plugins/coverbrowser.koplugin/covermenu.lua

```diff
@@ -238,25 +238,6 @@ function CoverMenu:updateItems(select_number)
                 -- And clear the rendering stack to avoid inheriting its dirty/refresh queue
                 UIManager:clearRenderStack()
 
-                -- Fudge the "Reset settings" button callback to also trash the cover_info_cache
-                local button = self.file_dialog.button_table:getButtonById("reset_settings")
```

Comment:
OK, this was getting the Button instance and tweaking it for nothing as its not the one that will be used.
Before #9571, we used to tweak the button definitiion in the buttons table (with the ugly indexes buttons[4][2]). getButtonById() allows us to get it by id instead, but it needs to work on a ButtonTable instance.
With this PR, we go tweaking the new Button instance later once it is made.

I somehow prefered all the tweaking to be done before the `self.file_dialog = ButtonDialogTitle:new{}` at bottom that we then just UIManager:show(). But I can't think of simpler way to do it (unless we add a local function to iterate the `buttons` definition to find stuff by id).
So, ok by me with just moving it like this PR.
Pinging @hius07.

Reply:
Yes, I've checked this out already, my mistake.

Since `orig_buttons` are preserved, another way to avoid references can be introducing in ButtonDialogTitle something similar to
https://github.com/koreader/koreader/blob/23c410344baba77f5b1a11b492a5da26864a31fd/frontend/ui/widget/radiobuttonwidget.lua#L193-L201